lib/syskit/cli/log_runtime_archive.rb

```diff
@@ -41,6 +52,33 @@ def process_root_folder
                 end
             end
 
+            # Manages folder available space
+            #
+            # The method will check if there is enough space to save more log files
+            # according to pre-established threshold.
+            #
+            # @param [integer] free_space_low_limit: required free space threshold, at
+            #   which the archiver starts deleting the oldest log files
+            # @param [integer] free_space_delete_until: post-deletion free space, at which
+            #   the archiver stops deleting the oldest log files
+            def ensure_free_space(free_space_low_limit: @free_space_low_limit,
+                free_space_delete_until: @free_space_delete_until)
+                stat = Sys::Filesystem.stat(@target_dir)
+                available_space = stat.bytes_free
+
+                return if available_space > free_space_low_limit
+
+                until available_space >= free_space_delete_until
+                    files = @target_dir.each_child.select(&:file?)
+                    break if files.empty?
```

Comment:
maybe warn that the folder is empty but the available space is smaller than the threshold?

Reply:
good call... raise an `ArgumentError`?